__tests__/reducers/history.spec.ts

```diff
@@ -0,0 +1,165 @@
+            payload: {
+                url: 'another-url'
+            }
+        } );
+
+        expect( newState ).not.toStrictEqual( state );
+        expect( newState ).toEqual( {
+            [date]: [
+                {
+                    url: 'safe://another-url',
+                    timeStamp: new Date().toLocaleTimeString()
+                }
+            ]
+        } );
+    } );
+    it( 'add another url', () => {
+        const state = {
+            [date]: [
+                {
+                    url: 'safe://another-url',
+                    timeStamp: '2:20:28 PM'
+                }
+            ]
+        };
+        const newState = history( state, {
+            type: TABS_TYPES.UPDATE_TAB_URL,
+            payload: {
+                url: 'another-another-url'
+            }
+        } );
+
+        expect( newState ).not.toStrictEqual( state );
+        expect( newState ).toEqual( {
+            [date]: [
+                {
+                    url: 'safe://another-another-url',
+                    timeStamp: new Date().toLocaleTimeString()
+                },
+                {
+                    url: 'safe://another-url',
+                    timeStamp: '2:20:28 PM'
+                }
+            ]
+        } );
+    } );
+
+    it( 'state should be the same if the same url is passed', () => {
+        const state = {
+            [date]: [
+                {
+                    url: 'safe://another-another-url',
+                    timeStamp: new Date().toLocaleTimeString()
+                },
+                {
+                    url: 'safe://another-url',
+                    timeStamp: '2:20:28 PM'
+                }
+            ]
+        };
+        const newState = history( state, {
+            type: TABS_TYPES.UPDATE_TAB_URL,
+            payload: {
+                url: 'another-another-url'
+            }
+        } );
+
+        expect( newState ).toStrictEqual( state );
+    } );
+
+    it( 'add pass a different url with a different date', () => {
+        const state = {
+            '5/21/2019': [
+                {
+                    url: 'safe://another-another-url',
+                    timeStamp: '4:45:69 PM'
+                },
+                {
+                    url: 'safe://another-url',
+                    timeStamp: '2:20:28 PM'
+                }
+            ],
+            [date]: [
+                {
+                    url: 'safe://hello',
+                    timeStamp: '2:25:29 PM'
+                }
+            ]
+        };
+
+        const newState = history( state, {
+            type: TABS_TYPES.UPDATE_TAB_URL,
+            payload: {
+                url: 'helloAgain'
+            }
+        } );
+
+        expect( newState ).not.toStrictEqual( state );
+        expect( newState ).toEqual( {
+            [date]: [
+                {
+                    url: 'safe://helloAgain',
+                    timeStamp: new Date().toLocaleTimeString()
```

Comment:
@manavbp are we store localeTimeStrings here? I think we should be using UTC in the store, and converting for display only. Otherwise we should be keeping the timezone info etc too.

UTC in the store seems simplest to me. And we can let the UI figure out displaying this info, and so we don't lose any data in the store itself. (We should accordingly check that a UTC or `Date` object is passed in the store and throw otherwise (and have tests for this too)